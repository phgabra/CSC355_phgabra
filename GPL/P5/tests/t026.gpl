
int b = 0;

int c = !b;
int d = -c;

int e = -b;
int f = !b;

int g = !-b;
int h = -!b;

int i = !(-b);
int j = -(!b);

// ! should have a higher precedence than * resulting in 2
// If the precedences are reverse the result would be 0
int k = !b + 1;

