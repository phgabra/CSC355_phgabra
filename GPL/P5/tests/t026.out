gpl.cpp::main()
  input file(tests/t026.gpl)
  random seed(42)
  read_keypresses_from_standard_input(false)
  dump_pixels(false)
  symbol_table(true)
  graphics(false)

gpl.cpp::main() Calling yyparse()


gpl.cpp::main() after call to yyparse().

No errors found (parser probably worked correctly).


Printing the symbol table from main()
int b = 0
int c = 1
int d = -1
int e = 0
int f = 1
int g = 1
int h = -1
int i = 1
int j = -1
int k = 2
Graphics is turned off by the Makefile.  Program exiting.

gpl.cpp::main() done.
